.github/workflows/update-pre-release-branches.yaml

```diff
@@ -0,0 +1,91 @@
+name: Auto-update pre-release branches
+
+on:
+  pull_request:
+    paths:
+      - .github/workflows/update-pre-release-branches.yaml
+  schedule:
+    # Run 20 minutes after midnight, giving the k8s-snap nightly job
+    # enough time to pick up new k8s releases and setup the git branches.
+    - cron: "20 0 * * *"
+
+permissions:
+  contents: read
+
+jobs:
+  determine:
+    name: "Determine k8s pre-release"
+    runs-on: ubuntu-latest
+    outputs:
+      preRelease: ${{ steps.determine.outputs.preRelease }}
+      gitBranch: ${{ steps.determine.outputs.gitBranch }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.BOT_SSH_KEY }}
+      - name: Harden Runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+      - name: Install Python dependencies
+        shell: bash
+        run: pip3 install -r ./scripts/requirements.txt
+      - name: Determine outstanding pre-release
+        id: determine
+        run: |
+          preRelease=`python3 ./scripts/k8s_releases.py get_outstanding_prerelease`
+          echo "preRelease=$preRelease" >> "$GITHUB_OUTPUT"
+
+          if [[ -n "$preRelease" ]]; then
+            gitBranch="autoupdate/$preRelease"
+          fi
+          echo "gitBranch=$gitBranch" >> "$GITHUB_OUTPUT"
+  handle-pre-release:
+    name: Handle pre-release ${{ needs.determine.outputs.preRelease }}
+    needs: [determine]
+    secrets: inherit
+    uses: ./.github/workflows/create-release-branch.yaml
+    if: ${{ needs.determine.outputs.preRelease }} != ''
+    with:
+      branches: ${{ needs.determine.outputs.gitBranch }}
+  clean-obsolete:
+    runs-on: ubuntu-latest
+    env:
+      CHARMCRAFT_AUTH : ${{ secrets.CHARMCRAFT_AUTH }}
+      LPCREDS_B64: ${{ secrets.LP_CREDS }}
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          ssh-key: ${{ secrets.BOT_SSH_KEY }}
+      - name: Harden Runner
+        uses: step-security/harden-runner@v2
+        with:
+          egress-policy: audit
+      - name: Setup Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.12'
+          cache: 'pip'
+      - name: Install Python dependencies
+        shell: bash
+        run: pip3 install -r ./scripts/requirements.txt
+      - name: Clean obsolete branches
+        run: |
```

Comment:
Just a comment for the follow-up PR for the cleanup:
This step should just be something like:
```
./scripts/k8s_release.py remove_obsolete_prereleases
```

We want to avoid having any logic in the Github CI were applicable.